File: core/server/sv_main.lua
RegisterServerEvent('saveObjectDataToFile')
AddEventHandler('saveObjectDataToFile', function(objectData)
    local fileToCreate = io.open('resources//vinObjDebugger//saved_object_data.txt', "a")
    local fileOutput = saveObjData(objectData)
    io.output(fileToCreate)
    io.write(fileOutput)
    io.close(fileToCreate)
end)

saveObjData = function(objectData)
    local saveToFileDataText = ""
    
    for i = 1, #objectData do
        local newDataInserted = {}

        for _, val in pairs(objectData[i]) do
            newDataInserted[#newDataInserted + 1] = val
        end

        saveToFileDataText = "Date Created: ".. os.date("!%Y-%m-%dT%H:%M:%SZ\n") .. json.encode(newDataInserted) .. "\n\n"
    end

    return saveToFileDataText
end

File: core/server/sv_main.js
RegisterNetEvent('saveObjectDataToFile');
on('saveObjectDataToFile', (objectData) => {
    let fileToCreate = fs.openSync(`${GetResourcePath(GetCurrentResourceName())}/saved_object_data.txt`, 'a');
    let fileOutput = saveObjData(objectData);
    fs.writeFileSync(fileToCreate, fileOutput);
    fs.closeSync(fileToCreate);
});

const saveObjData = (objectData) => {
    let saveToFileDataText = "";
    let newObjectDataInserted = {};
    
    let newTable = newArrayLoop(objectData);
    newObjectDataInserted = Array.from(newTable);
    
    saveToFileDataText = `Date Created: ${new Date().toISOString()}\nValues saved: ${JSON.stringify(newObjectDataInserted)} (Usage: [xPos, yPos, zPos, xRot, yRot, zRot])\n\n`;
    return saveToFileDataText;
}

const newArrayLoop = (tbl) => {
    let newTableInserted = [];
    for (let i = 0; i < tbl.length; i++) {
        newTableInserted.push(tbl[i]);
    }
    return newTableInserted;
}
